### ISSUE TITLE
Unable to use the nix-shell code: attribute pkg-config missing

### ISSUE THREAD

**Issue by ashkan-leo** [nix-community/poetry2nix#266]
I get the following error when using the example shell.nix
```
error: --- EvalError --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix-shell
at: (238:52) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix

   237|       nativeBuildInputs = old.nativeBuildInputs ++ [ pkgs.pkg-config ];
   238|       buildInputs = old.buildInputs ++ [ pkgs.hdf5 self.pkg-config self.cython ];
      |                                                    ^
   239|       configure_flags = "--hdf5=${pkgs.hdf5}";

attribute 'pkg-config' missing
(use '--show-trace' to show detailed location information)

```

My shell.nix is
```
{ pkgs ? import <nixpkgs> {} }:
let
  myAppEnv = pkgs.poetry2nix.mkPoetryEnv {
    projectDir = ./.;
    editablePackageSources = {
      my-app = ./src;
    };
  };
in
pkgs.mkShell {
  buildInputs = [ myAppEnv pkgs.hello ];
}
```

The stack trace:
```
error: --- EvalError --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix-shell
at: (238:52) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix

   237|       nativeBuildInputs = old.nativeBuildInputs ++ [ pkgs.pkg-config ];
   238|       buildInputs = old.buildInputs ++ [ pkgs.hdf5 self.pkg-config self.cython ];
      |                                                    ^
   239|       configure_flags = "--hdf5=${pkgs.hdf5}";

attribute 'pkg-config' missing
----------------------------------------------------------------------------------------------------------------- show-trace -----------------------------------------------------------------------------------------------------------------
trace: while evaluating anonymous lambda
at: (143:17) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/pkgs/stdenv/generic/make-derivation.nix

   142|           (map (drv: drv.__spliced.hostHost or drv) depsHostHost)
   143|           (map (drv: drv.crossDrv or drv) buildInputs)
      |                 ^
   144|         ]

trace: from call site
trace: while evaluating 'getOutput'
at: (464:23) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/lib/attrsets.nix

   463|   */
   464|   getOutput = output: pkg:
      |                       ^
   465|     if pkg.outputUnspecified or false

trace: from call site
trace: while evaluating the attribute 'buildInputs' of the derivation 'python3.8-h5py-3.1.0'
at: (111:5) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/pkgs/development/interpreters/python/mk-python-derivation.nix

   110|
   111|     name = namePrefix + name;
      |     ^
   112|

trace: while evaluating the attribute 'out.outPath'
at: (156:13) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/lib/customisation.nix

   155|             drvPath = assert condition; drv.${outputName}.drvPath;
   156|             outPath = assert condition; drv.${outputName}.outPath;
      |             ^
   157|           };

trace: while evaluating anonymous lambda
at: (643:24) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/lib/lists.nix

   642|    */
   643|  unique = foldl' (acc: e: if elem e acc then acc else acc ++ [ e ]) [];
      |                        ^
   644|

trace: from call site
at: (64:6) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/pkgs/top-level/python-packages.nix

    63|     modules = filter hasPythonModule drvs;
    64|   in unique ([python] ++ modules ++ concatLists (catAttrs "requiredPythonModules" modules));
      |      ^
    65|

trace: while evaluating 'requiredPythonModules'
at: (62:27) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/pkgs/top-level/python-packages.nix

    61|   # Get list of required Python modules given a list of derivations.
    62|   requiredPythonModules = drvs: let
      |                           ^
    63|     modules = filter hasPythonModule drvs;

trace: from call site
at: (80:33) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/pkgs/top-level/python-packages.nix

    79|         pythonPath = [ ]; # Deprecated, for compatibility.
    80|         requiredPythonModules = requiredPythonModules drv.propagatedBuildInputs;
      |                                 ^
    81|       };

trace: while evaluating 'requiredPythonModules'
at: (62:27) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/pkgs/top-level/python-packages.nix

    61|   # Get list of required Python modules given a list of derivations.
    62|   requiredPythonModules = drvs: let
      |                           ^
    63|     modules = filter hasPythonModule drvs;

trace: from call site
at: (80:33) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/pkgs/top-level/python-packages.nix

    79|         pythonPath = [ ]; # Deprecated, for compatibility.
    80|         requiredPythonModules = requiredPythonModules drv.propagatedBuildInputs;
      |                                 ^
    81|       };

trace: while evaluating 'requiredPythonModules'
at: (62:27) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/pkgs/top-level/python-packages.nix

    61|   # Get list of required Python modules given a list of derivations.
    62|   requiredPythonModules = drvs: let
      |                           ^
    63|     modules = filter hasPythonModule drvs;

trace: from call site
at: (15:13) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/pkgs/development/interpreters/python/wrapper.nix

    14|   env = let
    15|     paths = requiredPythonModules (extraLibs ++ [ python ] ) ;
      |             ^
    16|     pythonPath = "${placeholder "out"}/${python.sitePackages}";

trace: while evaluating the attribute 'passAsFile' of the derivation 'python3-3.8.7-env'
at: (7:7) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/pkgs/build-support/trivial-builders.nix

     6|     stdenv.mkDerivation ({
     7|       name = lib.strings.sanitizeDerivationName name;
      |       ^
     8|       inherit buildCommand;

trace: while evaluating the attribute 'buildInputs' of the derivation 'nix-shell'
at: (28:3) in file: /nix/store/a1z821kfag7ywd531bfkis5br6av9pwx-nixpkgs-21.03pre266131.e61999b3f24/nixpkgs/pkgs/build-support/mkshell/default.nix

    27| stdenv.mkDerivation ({
    28|   name = "nix-shell";
      |   ^
    29|   phases = ["nobuildPhase"];


```

**Comment by adisbladis** [comment]
This was fixed in 01c04202.
If you just update your nixpkgs version you will get a newer version of this override.

**Comment by ashkan-leo** [comment]
Thanks for your reply. I upgraded nix and that error went away.

Now, I have another error:

```
builder for '/nix/store/v7k9dxd336s6qzck37p5f6inmaxi5l4s-python3.8-h5py-3.2.1.drv' failed with exit code 1; last 10 log lines:
  patching sources
  configuring
  no configure script, doing nothing
  WARNING: The directory '/homeless-shelter/Library/Caches/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help

  error: invalid command 'configure'

```

I don't know why I'm getting this error. `poetry update` and `poetry shell` work as expected so I don't think this is an issue with the h5py package. I'm not sure where to ask this question either. Any help is appreciated!

Thanks!

**Comment by adisbladis** [comment]
It turns out our `h5py` overrides were outdated.
Fixed in 5247b36e8e9024d804f73eef40a754052d93baec.

**Comment by ashkan-leo** [comment]
Thanks for the fix. I got passed that error and h5py is successfully installed. 
Now I get a new error:

1. A lot is going on, but the main issue seems to be about psutil not finding the Xcode SDK files. I have verified installation of the XCode Command Line Developer Tools. I don't know why they can't be found.
2. I don't know what to make of the `TypeError: 'color' is an invalid keyword argument for print()` exception.

```
  creating build/temp.macosx-10.6-x86_64-3.8/psutil/arch/osx
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -DPSUTIL_POSIX=1 -DPSUTIL_SIZEOF_PID_T=4 -DPSUTIL_VERSION=580 -DPSUTIL_OSX=1 -I/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/include/python3.8 -c psutil/_psutil_common.c -o build/temp.macosx-10.6-x86_64-3.8/psutil/_psutil_common.o
  clang-7: warning: argument unused during compilation: '-fno-strict-overflow' [-Wunused-command-line-argument]
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -DPSUTIL_POSIX=1 -DPSUTIL_SIZEOF_PID_T=4 -DPSUTIL_VERSION=580 -DPSUTIL_OSX=1 -I/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/include/python3.8 -c psutil/_psutil_posix.c -o build/temp.macosx-10.6-x86_64-3.8/psutil/_psutil_posix.o
  clang-7: warning: argument unused during compilation: '-fno-strict-overflow' [-Wunused-command-line-argument]
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -DPSUTIL_POSIX=1 -DPSUTIL_SIZEOF_PID_T=4 -DPSUTIL_VERSION=580 -DPSUTIL_OSX=1 -I/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/include/python3.8 -c psutil/_psutil_osx.c -o build/temp.macosx-10.6-x86_64-3.8/psutil/_psutil_osx.o
  clang-7: warning: argument unused during compilation: '-fno-strict-overflow' [-Wunused-command-line-argument]
  psutil/_psutil_osx.c:38:10: fatal error: 'IOKit/IOKitLib.h' file not found
  #include <IOKit/IOKitLib.h>
           ^~~~~~~~~~~~~~~~~~
  1 error generated.
  Traceback (most recent call last):
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/unixccompiler.py", line 127, in _compile
      self.spawn(compiler_so + cc_args + [src, '-o', obj] +
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/ccompiler.py", line 910, in spawn
      spawn(cmd, dry_run=self.dry_run)
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/spawn.py", line 36, in spawn
      _spawn_posix(cmd, search_path, dry_run=dry_run)
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/spawn.py", line 157, in _spawn_posix
      raise DistutilsExecError(
  distutils.errors.DistutilsExecError: command 'clang' failed with exit status 1

  During handling of the above exception, another exception occurred:

  Traceback (most recent call last):
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/nix/store/ik60y0br4jkxlkp1jsp9nacq2ryy6dgl-python3.8-wheel-0.35.1/lib/python3.8/site-packages/wheel/bdist_wheel.py", line 290, in run
      self.run_command('build')
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/command/build.py", line 135, in run
      self.run_command(cmd_name)
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/nix/store/n2vvpg98jb81653m7cxgrcm2x5765n8d-python3.8-setuptools-50.3.1/lib/python3.8/site-packages/setuptools/command/build_ext.py", line 79, in run
      _build_ext.run(self)
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/command/build_ext.py", line 340, in run
      self.build_extensions()
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/command/build_ext.py", line 449, in build_extensions
      self._build_extensions_serial()
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/command/build_ext.py", line 474, in _build_extensions_serial
      self.build_extension(ext)
    File "/nix/store/n2vvpg98jb81653m7cxgrcm2x5765n8d-python3.8-setuptools-50.3.1/lib/python3.8/site-packages/setuptools/command/build_ext.py", line 196, in build_extension
      _build_ext.build_extension(self, ext)
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/command/build_ext.py", line 528, in build_extension
      objects = self.compiler.compile(sources,
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/ccompiler.py", line 574, in compile
      self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/unixccompiler.py", line 130, in _compile
      raise CompileError(msg)
  distutils.errors.CompileError: command 'clang' failed with exit status 1

  During handling of the above exception, another exception occurred:

  Traceback (most recent call last):
    File "/private/tmp/nix-build-python3.8-psutil-5.8.0.drv-0/pip-req-build-r3dmjf5a/setup.py", line 411, in main
      setup(**kwargs)
    File "/nix/store/n2vvpg98jb81653m7cxgrcm2x5765n8d-python3.8-setuptools-50.3.1/lib/python3.8/site-packages/setuptools/__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "/nix/store/5vn5f6slfg1nw8q0bhapd3vgk9j7559i-python3-3.8.7/lib/python3.8/distutils/core.py", line 163, in setup
      raise SystemExit("error: " + str(msg))
  SystemExit: error: command 'clang' failed with exit status 1

  During handling of the above exception, another exception occurred:

  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/private/tmp/nix-build-python3.8-psutil-5.8.0.drv-0/pip-req-build-r3dmjf5a/setup.py", line 436, in <module>
      main()
    File "/private/tmp/nix-build-python3.8-psutil-5.8.0.drv-0/pip-req-build-r3dmjf5a/setup.py", line 422, in main
      print(hilite("XCode (https://developer.apple.com/xcode/) "
  TypeError: 'color' is an invalid keyword argument for print()
  ----------------------------------------
  ERROR: Failed building wheel for psutil
  Running setup.py clean for psutil
Successfully installed python-constraint-1.4.0
Installing collected packages: pybind11
Successfully installed pybind11-2.6.2
/private/tmp/nix-build-python3.8-python-constraint-1.4.0.drv-0/python-constraint-1.4.0
Finished executing pipInstallPhase
post-installation fixup
strip is /nix/store/kiclfrcdg2lxifq68q7d354dlaqk8136-cctools-binutils-darwin-949.0.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/fh5m3n31sb0gfvlfyb80kxwcpn370l96-python3.8-python-constraint-1.4.0/lib
/private/tmp/nix-build-python3.8-pybind11-2.6.2.drv-0/pybind11-2.6.2
Finished executing pipInstallPhase
post-installation fixup
Building wheels for collected packages: sympy
strip is /nix/store/kiclfrcdg2lxifq68q7d354dlaqk8136-cctools-binutils-darwin-949.0.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/q66sjbhxs1ahc4f8bb8z5gqp7qcly63b-python3.8-pybind11-2.6.2/lib  /nix/store/q66sjbhxs1ahc4f8bb8z5gqp7qcly63b-python3.8-pybind11-2.6.2/bin
  Building wheel for pyrsistent (setup.py) ... done
  Created wheel for pyrsistent: filename=pyrsistent-0.17.3-cp38-cp38-macosx_10_12_x86_64.whl size=68607 sha256=6414b1e5c84ce416376e83144b3625ac1356d5b25063d88e318569ee2ea2c55f
  Stored in directory: /private/tmp/nix-build-python3.8-pyrsistent-0.17.3.drv-0/pip-ephem-wheel-cache-1bdade9v/wheels/60/17/c4/48b6b7da1f27a4e8e5854eb5a58a3779cddbf47393fbc872c0
Successfully built pyrsistent
Finished creating a wheel...
Finished executing pipBuildPhase
installing
Executing pipInstallPhase
/private/tmp/nix-build-python3.8-pyrsistent-0.17.3.drv-0/pyrsistent-0.17.3/dist /private/tmp/nix-build-python3.8-pyrsistent-0.17.3.drv-0/pyrsistent-0.17.3
Failed to build psutil
ERROR: Failed to build one or more wheels
error: --- Error --- nix-daemon
builder for '/nix/store/qqy69wmps6w347blsw5n42qjl443m036-python3.8-psutil-5.8.0.drv' failed with exit code 1; last 10 log lines:
      File "/private/tmp/nix-build-python3.8-psutil-5.8.0.drv-0/pip-req-build-r3dmjf5a/setup.py", line 436, in <module>
        main()
      File "/private/tmp/nix-build-python3.8-psutil-5.8.0.drv-0/pip-req-build-r3dmjf5a/setup.py", line 422, in main
        print(hilite("XCode (https://developer.apple.com/xcode/) "
    TypeError: 'color' is an invalid keyword argument for print()
    ----------------------------------------
    ERROR: Failed building wheel for psutil
    Running setup.py clean for psutil
  Failed to build psutil
  ERROR: Failed to build one or more wheels
```
